File: Blind75/Binary Search Tree Kth Smallest Element/Binary_Search_Tree_Kth_Smallest_Element.py
def kthSmallestElementInABst(root, k) :
    sorted_array = []
    
    def traversal(curr):
        if curr is None : return
        
        if curr.left: 
            traversal(curr.left)
            
        sorted_array.append(curr.val)
        traversal(curr.right)
        
    traversal(root)
    return sorted_array[k - 1]

File: Blind75/Binary Search Tree Kth Smallest Element/Binary_Search_Tree_Kth_Smallest_Element.js
function kthSmallestElementInABst(root, k) {
    // traversing 
    // left -> root -> right
    let sortedArray = [];

    function traversing(curr) {
        if (!curr) return;
        if (curr.left) {
            traversing(curr.left);
        }
        sortedArray.push(curr.val);
        traversing(curr.right);
    }

    traversing(root);

    // BST gives sorted order after traversing
    return sortedArray[k - 1];
}
